src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -513,10 +514,45 @@ static bool match_sysname(sd_device_enumerator *enumerator, const char *sysname)
         return false;
 }
 
+static int match_initialized(sd_device_enumerator *enumerator, sd_device *device) {
+        int r;
+
+        assert(enumerator);
+        assert(device);
+
+        if (enumerator->match_initialized < 0)
+                return true;
+
+        r = sd_device_get_is_initialized(device);
+        if (r < 0) {
+                /* this is necessarily racey, so ignore missing devices */
+                if (r != -ENOENT)
+                        return r;
+
+                return false;
+        }
+
+        if (enumerator->match_initialized == r)
```

Comment:
given that both of these are `int`, and not `bool`, but you actually want a boolean comparison here, i'd write this as `if (!enumerator->match_initialized == !r)`

of course, if everything is alright the only values stored in these variables are 0 + 1, but I think it's useful to communicate the boolean concept here to the reader